CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+cmake_minimum_required(VERSION "3.19.2")
+
+include(${CMAKE_SOURCE_DIR}/common_macros.cmake)
+
+project(state-trafficlight VERSION 1.0 LANGUAGES C)
+
+SET(TARGET_NAME state-trafficlight)
+message(STATUS "Value of CMAKE_CURRENT_BINARY_DIR: ${CMAKE_CURRENT_BINARY_DIR}")
+message(STATUS "Value of CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
+
+
+include_directories(${PROJECT_SOURCE_DIR}/common/include)
+
+add_library(common_lib STATIC
+            common/src/traffic_light.c
+            common/src/math_operations.c           
+)
+
```

Comment:
The -Wall compiler flag enables all compiler warning messages. Wondering if you can enable -Wall through the CMake build system so we get the warning messages during compilation.